server/controllers/user.ts

```diff
@@ -1,40 +1,48 @@
 import express from 'express'
 import { User } from '../models/user'
+import {Avatar} from '../models/avatar'
 
 
 async function getUserData(req: express.Request, res: express.Response) {
     try {
-        const { email } = req.oidc.user;
+        const  {email,sub} = req.oidc.user
+        console.log('request opened to retrieve information from: ',email)
         const user = await User.findOne({ email: email })
         if (user) {
-            res.status(201).json({ "statusCode": 200, "message": "Retrieving information of the user", user: req.oidc.user });
+            const resUser = await User.findOne({ email: email }).populate('avatar');
+            res.status(201).json({ "statusCode": 201, "message": "Retrieving information of the user", user: resUser });
         } else {
+            const username = email.substr(0,email.indexOf('@'));
+            const avatar = await Avatar.findOne({contentType: 'monkey.png'});
             const newUser = new User({
-                email: req.oidc.user.email,       
+                id: sub,
+                email: email,
+                username: username,
+                avatar:avatar.id, 
             });
-            console.log(newUser)
-            const user = await newUser.save();
-
-            res.status(201).json({ "statusCode": 200, "message": "User created", user });
+            await newUser.save();
+            const resUser = await User.findOne({ email: email }).populate('avatar');
+            res.status(201).json({ "statusCode": 201, "message": "User created", user: resUser });
         }
     } catch (error) {
         console.log(error);
-        res.status(500).json({ "statusCode": 200, "message": error });
+        res.status(500).json({ "statusCode": 500, "message": error });
     }
 }
 
 async function updateUsername(req: express.Request, res: express.Response) {
     try {
-        const username = req.body.username;
+        const {username} = req.body;
+        console.log(username);
         const { email } = req.oidc.user;
```

Comment:
I presume we can do this because we've already checked the user is logged in?

Reply:
Yes, auth0 autheticates the user.